### ISSUE TITLE
Litecoin improvement proposal - Creating a web wallet

### ISSUE THREAD

**Issue by PWKad** [litecoin-project/litecoin#396]
## TL:DR

The wallet released with litecoin core (and of course bitcoin) is due for an upgrade.  Currently the wallet is very outdated and [redesign has been proposed](https://medium.com/litecoin-foundation/redesigning-litecoin-core-with-lightning-capabilities-3a8ebbea590a).  

The proposal shows some modern design techniques which will be difficult to implement in the existing wallet.  This proposal outlines an updated web-based wallet running on top of the node.  Long term this would be hopefully a proposal to make back to Bitcoin core to replace the existing wallet but for now assume this is ancillary to core.

## Front-end

The wallet will be written in modern JavaScript and HTML with SASS for the styling.  Initially this will be written with Aurelia ([a JavaScript framework](http://aurelia.io/)) for speed of development and because it is most spec compliant and would be easiest to port to any other framework or back to vanilla.js.  Also because I know a large community of developers would are able to support this endeavor.

## Mid-tier

To handle abstracting out interactions with the node a simple node.js API can be used to proxy requests to the node.  This should speed up development as well.  I'm fairly lenient on this portion of the proposal.

## Timeline / Milestones

1. Design - design assets created - done
2. Develop - develop working proof of concept - in progress ~ complete by December 1 2017
3. Review / Approval of functionality by Litecoin core maintainers
4. Develop - Start to Productionize the wallet
5. Submit a production-ready PR
6. Submit a proposal back to Bitcoin core for the same wallet

## Requirements

- Internationalization support (ex. English / Spanish)
- Accessible (ex. Screen reader friendly)
- ?

Feedback welcome, and if this is the wrong place to open a discussion like this or if this doesn't seem like a proposal that makes sense please feel free to close this out.

**Comment by PWKad** [comment]
Instead of responding with support that they would support this project from a development standpoint (questions, code contributions, etc...) I'm asking the Aurelia core team and community members to add the :hooray: reaction or plus one it to show support.

**Comment by coblee** [comment]
Have you seen this? https://medium.com/litecoin-foundation/redesigning-litecoin-core-with-lightning-capabilities-3a8ebbea590a

**Comment by PWKad** [comment]
@coblee Yes I actually linked that in the first sentence of my proposal as the design work.

**Comment by coblee** [comment]
Haha, my bad! I like the idea of a web-based UI for the wallet functionality. Thanks for helping out!

**Comment by TylerJPresley** [comment]
I'd love to contribute somehow to this.

**Comment by PWKad** [comment]
@TylerJPresley and anyone else who is interested in contributing, proof of concept checklist - https://github.com/PWKad/litecoin-core-wallet/issues/2

**Comment by VKoskiv** [comment]
I would like to get someone more knowledgeable to comment on the security risks of running a node.js based frontend for a crypto wallet.
I'm no expert, but there is a history of [malicious packages](https://duo.com/blog/hunting-malicious-npm-packages) on NPM.

Is there more specific reasoning why the UI overhaul cannot be performed using QT?

**Comment by sriramgopal** [comment]
Hi,

I have an interesting idea for Litecoin. How can I bring this to the attention of devs?

**Comment by VKoskiv** [comment]
@sriramgopal Open a new issue.

**Comment by sriramgopal** [comment]
@VKoskiv It is not a technical idea. It's more of a business model and I'm unsure if I can post such stuff here.

**Comment by PWKad** [comment]
@VKoskiv sorry for not responding earlier, if someone is willing and experienced to make the updates in QT I think that would be good as well.  Personally my opinion is that front-end development has progressed well past what QT offers and JavaScript is my primary language so it's a natural fit.

**Comment by losh11** [comment]
Closing due to inactivity.
